crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -220,12 +222,18 @@ class C:
         # TODO: this redeclaration should be an error
         self.y: str = "a"
 
+        # TODO: this redeclaration should be an error
+        self.z: str = "a"
+
 c_instance = C()
 
 reveal_type(c_instance.x)  # revealed: Unknown | int | str
 
 # TODO: We should probably infer `int | str` here.
 reveal_type(c_instance.y)  # revealed: int
+
+# TODO: We should probably infer `int | str` here.
+reveal_type(c_instance.z)  # revealed: int
```

Comment:
Since we'll have emitted a diagnostic above complaining about the redeclaration, I think the current behaviour here is also reasonable TBH. I wouldn't worry too much about inferring a union here if it complicates the implementation.